jupyter_dash/jupyter_app.py

```diff
@@ -327,7 +333,17 @@ def run():
             wait_exponential_max=1000
         )
         def wait_for_app():
-            res = requests.get(alive_url).content.decode()
+            try:
+                err = err_q.get_nowait()
+                if err:
+                    raise err
+            except queue.Empty:
+                pass
+            req = requests.get(alive_url)
+            res = req.content.decode()
+            if req.status_code == 500:
```

Comment:
Two questions here:
(1) Any chance we'd fail with a code other than 500? Maybe switch to `if req.status_code != 200`?
(2) Is there any way an error would make it into the queue later, ie during the `requests.get` call? Should we check the queue again after receiving this 500 response (or even check always after `requests.get`?

Reply:
(1) The error handler always return 500, that catches the errors in the `before_first_request` in dash and layout/deps errors. I don't think in dash we return other status code, but there could be with plugin.
(2) The queue get the error from server start, the requests error goes into Exception handler to return a 500, if the server failed to start the `requests.get` will raise connection error, next retry it get queue error if it wasn't there. Think if there a requests error it should check the queue after to ensure no race condition to get the proper error on the last retry.